src/coldcard.js

```diff
@@ -507,107 +516,177 @@ export class ColdcardSignMultisigTransaction extends ColdcardInteraction {
+            validateRootFingerprint(key);
+
+            // get the network from xpub prefix
+            const prefix = value.slice(0, 4)
+            
+            const network = getNetworkFromPrefix(prefix);
+
+            if (options.network) {
+              assert(network === options.network, 'Extended public key networks from config do not match.');
+            }
+            options.network = network;
+
+            if (!options.extendedPublicKeys) {
+              options.extendedPublicKeys = [];
+            }
+            options.extendedPublicKeys.push({
+              xfp: key,
+              xpub: value,
+            })
+          } catch(e) {
+            // TODO: Should we handle this error or 
+            // allow for unknown keys/values? Don't know if this is an incorrect xpub
+            // or some unrecognized key;
+            break;
+          }
+          break;
         }
-        if (isNaN(Number(`0x${xpub.xfp}`))) {
-          throw new Error("XFP is invalid hex");
-        }
-        return true;
-      })) {
-      this.extendedPublicKeys = this.jsonConfig.extendedPublicKeys;
-    } else {
-      throw new Error("Configuration file needs extendedPublicKeys.");
-    }
+      }
+    })
+    
+    return new MultisigWalletConfig(options);
+}
+
+/**
+ * @description returns a config in a ColdCard compatible format
+ * @param {object|module:config.MultisigWalletConfig} _config - config instance
+ * or options object for creating a valid config to generate coldcard file fro
+ * @returns {string} output - config as a string compatible with coldcard
+ */
+export function generateColdcardConfig(_config) {
+  let config = _config
+  // make sure we have a valid config object or can create one from the argument
+  if (!(config instanceof MultisigWalletConfig)) {
+    assert(typeof config === 'object', 'Must pass an object or MultisigWalletConfig to generate Coldcard config');
+    config = new MultisigWalletConfig(_config)
   }
+    // verify extended public keys and their fingerprints
+    config.validateExtendedPublicKeys(true);
 
-  /**
-   * @returns {string} output to be written to a text file and uploaded to Coldcard.
-   */
-  adapt() {
     // Coldcard configs can't have spaces in the names, it just splits on space and takes the first word.
     // Currently operating without derivation paths per xpub until feature is added.
     let output = `# Coldcard Multisig setup file (exported from unchained-wallets)
 # https://github.com/unchained-capital/unchained-wallets
 # v${COLDCARD_WALLET_CONFIG_VERSION}
 # 
-Name: ${this.name}
-Policy: ${this.requiredSigners} of ${this.totalSigners}
-Format: ${this.addressType}
+Name: ${config.name}
+Policy: ${config.requiredSigners} of ${config.totalSigners}
+Format: ${config.addressType}
 
 `;
     // We need to loop over xpubs and output `xfp: xpub` for each
-    let xpubs = this.extendedPublicKeys.map((xpub) => `${xpub.xfp}: ${xpub.xpub}`);
```

Comment:
obviously will need an update after #49 and firmware version `3.2.1` from coldcard